File: server/router/website.go
package router

import (
	"server/api"

	"github.com/gin-gonic/gin"
)

type WebsiteRouter struct {
}

func (w *WebsiteRouter) InitWebsiteRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	websiteRouter := Router.Group("website")
	websitePublicRouter := PublicRouter.Group("website")

	websiteApi := api.ApiGroupApp.WebsiteApi
	{
		websiteRouter.POST("createFooterLink", websiteApi.WebsiteCreateFooterLink)
		websiteRouter.DELETE("deleteFooterLink", websiteApi.WebsiteDeleteFooterLink)
	}
	{
		websitePublicRouter.GET("logo", websiteApi.WebsiteLogo)
		websitePublicRouter.GET("title", websiteApi.WebsiteTitle)
		websitePublicRouter.GET("info", websiteApi.WebsiteInfo)
		websitePublicRouter.GET("footerLink", websiteApi.WebsiteFooterLink)
		websitePublicRouter.GET("yiyan", websiteApi.WebsiteYiyan)
	}
}

File: web/src/api/website.ts
import type {ApiResponse} from "@/utils/request";
import service from "@/utils/request";
import type {Website} from "@/api/config";

export const websiteInfo = (): Promise<ApiResponse<Website>> => {
    return service({
        url: '/website/info',
        method: 'get',
    })
}

export interface FooterLink {
    title: string;
    link: string;
}

export const websiteFooterLink = (): Promise<ApiResponse<FooterLink[]>> => {
    return service({
        url: '/website/footerLink',
        method: 'get',
    })
}

export const websiteCreateFooterLink = (data: FooterLink): Promise<ApiResponse<undefined>> => {
    return service({
        url: '/website/createFooterLink',
        method: 'post',
        data: data,
    })
}

export const websiteDeleteFooterLink = (data: FooterLink): Promise<ApiResponse<undefined>> => {
    return service({
        url: '/website/deleteFooterLink',
        method: 'delete',
        data: data,
    })
}

export const websiteYiyan = (): Promise<ApiResponse<string>> => {
    return service({
        url: '/website/yiyan',
        method: 'get',
    })
}
